src/main/java/jenkins/plugins/slack/workflow/SlackResponse.java

```diff
@@ -0,0 +1,39 @@
+package jenkins.plugins.slack.workflow;
+
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.plugins.scriptsecurity.sandbox.whitelists.Whitelisted;
+import org.json.JSONObject;
+
+import java.io.Serializable;
+
+public class SlackResponse implements Serializable {
+    private String channelId;
+    private String ts;
+
+    public SlackResponse(String slackResponseString) {
+        if (StringUtils.isEmpty(slackResponseString)) {
```

Comment:
I don't think this change is quite what you wanted 😉 
1. "ok" is the text returned for non bot users, but yes its correctly null as your new test demonstrates
2. now the case that you want doesn't work as this code is only invoked when the response is empty

maybe 
```suggestion
        if (!"ok".equals(slackResponseString)) { // non botUsers respond with only 'ok'
```

Reply:
Whoops, I seem to have mixed up two different approaches somehow... thanks for your patience.

I was trying to make the response string only populate when bot user is turned on, and in other cases return a SlackResponse object with null fields. I would be worried about any other unknown cases if I simply checked for "ok".